### DISCUSSION TITLE
cpanm --installdeps equivalent

### DISCUSSION THREAD

**Discussion by waterkip** [skaji/cpm#227] [Q&A]
Hello, I'm looking for an equivalent of `cpanm --installdeps Pkg::Foo`. Does cpm offer such a thing?

**Comment by rwp0** [comment]
```
# from cpanfile (with cpanfile.snapshot if any)
> cpm install
```

https://metacpan.org/dist/App-cpm/view/script/cpm

just run the `install` command.

---

Not sure of the `Pkg::Foo` part.

**Reply by waterkip** [reply]
The thing is, I can now say with cpanm: `cpanm --installdeps Pkg::Foo` and it installs the deps of Pkg::Foo, but does not install Pkg::Foo itself. We have some build instructions where we want to test as much as possible before install but the Pkg::Foo has to be installed without tests for reasons.

I would like to make the switch to cpm, but without the option to install deps alone I cannot make the switch.

**Comment by skaji** [comment]
Currently there is no such option.

**Comment by zmughal** [comment]
I would also like a way of doing this. My use case is that I might have build options / env variables that I want to pass into a particular dist, but I want all of its dependencies to install using their defaults.
